test/external/testcorenrn/CMakeLists.txt

```diff
@@ -0,0 +1,118 @@
+    MODFILE_PATTERNS mod/*.mod
+    SCRIPT_PATTERNS common.hoc defvar.hoc test${test}.hoc
+    OUTPUT asciispikes::out${test}.dat)
+  # Set up a test that runs the simulation using NEURON.
+  nrn_add_test(
+    GROUP testcorenrn_${test}
+    NAME neuron
+    PROCESSORS ${num_ranks}
+    COMMAND
+      OMP_NUM_THREADS=1 HOC_LIBRARY_PATH=. ${MPIEXEC_NAME} ${MPIEXEC_NUMPROC_FLAG} ${num_ranks}
+      ${MPIEXEC_PREFLAGS} special ${MPIEXEC_POSTFLAGS} -mpi -c sim_time=100 test${test}.hoc)
+  # Set up a test that runs the simulation using CoreNEURON's online mode
+  if(NOT "${test}" STREQUAL "patstim")
+    # Exclude the patstim test from running via CoreNEURON online mode
+    nrn_add_test(
+      GROUP testcorenrn_${test}
+      NAME coreneuron_online
+      REQUIRES coreneuron
+      PROCESSORS ${num_ranks}
+      COMMAND
+        OMP_NUM_THREADS=1 HOC_LIBRARY_PATH=. ${MPIEXEC_NAME} ${MPIEXEC_NUMPROC_FLAG} ${num_ranks}
+        ${MPIEXEC_PREFLAGS} special ${MPIEXEC_POSTFLAGS} -mpi -c sim_time=100 -c coreneuron=1
+        test${test}.hoc)
+  endif()
+
+  set(extra_args)
+  if("${test}" STREQUAL "patstim")
+    list(APPEND extra_args --pattern patstim.spk)
+  endif()
+  # Set up a test that runs the simulation using CoreNEURON's offline mode. First we must run NEURON
+  # and tell it to dump the model without simulating it.
+  nrn_add_test(
+    GROUP testcorenrn_${test}
+    NAME coreneuron_offline
+    REQUIRES coreneuron
+    PROCESSORS ${num_ranks}
+    PRECOMMAND OMP_NUM_THREADS=1
+               HOC_LIBRARY_PATH=.
+               ${MPIEXEC_NAME}
+               ${MPIEXEC_NUMPROC_FLAG}
+               ${num_ranks}
+               ${MPIEXEC_PREFLAGS}
+               special
+               -mpi
+               -c
+               sim_time=100
+               -c
+               dumpmodel=1
+               test${test}.hoc
+    COMMAND
+      OMP_NUM_THREADS=1 ${MPIEXEC_NAME} ${MPIEXEC_NUMPROC_FLAG} ${num_ranks} ${MPIEXEC_PREFLAGS}
+      special-core ${MPIEXEC_POSTFLAGS} -d coredat --mpi -e 100 ${extra_args}
+    OUTPUT asciispikes::out.dat)
+  nrn_add_test_group_comparison(
+    GROUP testcorenrn_${test}
+    REFERENCE_OUTPUT asciispikes::external/tests/testcorenrn/reference/out_${test}.spk)
+endforeach()
+
+# Run the direct_tests that compare internally NEURON and CoreNEURON simulations
+foreach(test ${direct_tests})
+  if(DEFINED mpi_ranks_${test})
+    set(num_ranks ${mpi_ranks_${test}})
+  else()
+    set(num_ranks 1)
+  endif()
+  nrn_add_test_group(
+    NAME testcorenrn_${test}
+    SUBMODULE tests/testcorenrn
+    MODFILE_PATTERNS mod/*.mod
+    SCRIPT_PATTERNS common.hoc defvar.hoc test${test}.hoc
+    OUTPUT asciispikes::out${test}.dat)
+  nrn_add_test(
+    GROUP testcorenrn_${test}
+    NAME direct
+    REQUIRES coreneuron
+    PROCESSORS ${num_ranks}
+    COMMAND
+      HOC_LIBRARY_PATH=. ${MPIEXEC_NAME} ${MPIEXEC_NUMPROC_FLAG} ${num_ranks} ${MPIEXEC_PREFLAGS}
+      special ${MPIEXEC_POSTFLAGS} -mpi -c sim_time=100 test${test}.hoc)
```

Comment:
I *think* depending on based on MPI or non-MPI build, the argument like `-mpi` need to be removed.

Reply:
discussed offline - no need to worry about non-mpi build